### ISSUE TITLE
[🚀 Feature]: Append Basic Auth config in /opt/selenium/config.toml with official Helm Chart

### ISSUE THREAD

**Issue by amardeep2006** [SeleniumHQ/docker-selenium#1935]
### Feature and motivation

I am exploring a way to autopopulate Basic Auth information in Hub and Router containers with existing helm charts.

I have looked at PR https://github.com/SeleniumHQ/docker-selenium/pull/1894 which allows us to set ROUTER_USERNAME and ROUTER_PASSWORD in the environment variable of Hub/Router containers. 

Can the Docker image startup scripts for Hub and Router be modified to detect the presence of ROUTER_USERNAME and ROUTER_PASSWORD OS variables and append router username and password in /opt/selenium/config.toml file if variables are present ?

### Usage example

Based on my understanding basic auth can not be enabled in Hub deployed in Kubernetes with default selenium-grid helm chart. If above idea sounds feasible, it may enable basic auth without need of modifying default helm chart and adding config maps.
Please correct me if my understanding is wrong and there is an easy way to enable basic auth in kubernetes.

**Comment by diemol** [comment]
>Can the Docker image startup scripts for Hub and Router be modified to detect the presence of ROUTER_USERNAME and ROUTER_PASSWORD OS variables and append router username and password in /opt/selenium/config.toml file if variables are present ?

Do you think this is a good idea? The values will be written to a file and shown in the log that can be read and potentially used to access the Grid. 

I am not following. The helm chart already allows this. What are you missing?

**Comment by amardeep2006** [comment]
@diemol Thanks for the quick response . I know helm chart allows setting up ROUTER_USERNAME and ROUTER_PASSWORD for the grid during the deployment . Let me rephrase my question : 
How to configure Hub/ router container so that the Grid Needs basic authentication to access. 

May be I am missing something very obvious.   My understanding is Hub/Router container will run without authentication unless below config is present in /opt/selenium/config.toml 
```
[router]
username = "admin"
password = "myStrongPassword"
```

**Comment by diemol** [comment]
Not at all, you can set those environment variables on the Hub and it will work.

**Comment by amardeep2006** [comment]
Thanks Let me try and share results.

**Comment by amardeep2006** [comment]
You are correct @diemol . Interestingly, I could not find details of ENV VARs  ROUTER_USERNAME and ROUTER_PASSWORD in Selenium documentation . Google Search does return a page that talks about it.
https://www.webelement.click/en/selenium_grid_4_complete_guide_to_configuration_flags#_router_flags

Test case do work on selenium-grid.local/wd/hub but I cannot login to ui  I am getting 401 when I open the url 
http://selenium-grid.local/ui  . Should it not popup for username and password ?

![image](https://github.com/SeleniumHQ/docker-selenium/assets/50517840/95b5c8aa-881b-4ce6-851c-72ad17249e1f)

Thanks for your time.

**Comment by diemol** [comment]
It is giving you a 401, and if you did not get the prompt, it can be a configuration in Kubernetes.

**Comment by amardeep2006** [comment]
The server is sending back the basic auth challenge but seems browser is not respecting that . 
I even tried deploying in cluster that has ssl enabled and with nginx ingress. Still same results 

Local kubernetes with  traefik ingress : 
![image](https://github.com/SeleniumHQ/docker-selenium/assets/50517840/99d1544f-8b64-42b2-a7c4-38a1c02f7a4f)

Server with ssl enabled and nginx ingress :  
![image](https://github.com/SeleniumHQ/docker-selenium/assets/50517840/584de45f-de84-44a2-825c-7fd56ad57374)



I am using grid version 4.12.1  . I am ready to raise a new bug if you think it is a bug.

**Comment by diemol** [comment]
It seems to be either an issue with your browser or the networking config. Several people use auth and we do not have any issues reported.

**Comment by amardeep2006** [comment]
Thanks a lot @diemol  for your patience and building this amazing platform. The issue was my corporate browser policies. Probably something is blocking it on Windows Machine.

It is working when I opened grid url inside a Google Chrome on a Linux Machine ( browser inside selenium/standalone-chrome:4.12.1-20230904). 
![image](https://github.com/SeleniumHQ/docker-selenium/assets/50517840/174eff0c-f2e3-4508-bcef-f03f55c0a784)


The feature is already implemented hence closing the same.
